interfaces/functions.yaml

```diff
@@ -856,6 +859,9 @@ igraph_has_loop:
 igraph_has_multiple:
     PARAMS: GRAPH graph, OUT BOOLEAN res
 
+igraph_count_loops:
+    PARAMS: GRAPH graph, OUT INTEGER loop_count
```

Comment:
Can we imagine a failure mode for `igraph_count_loops()` for now or the future? If we can't, we can make `loop_count` a return value just like we do with `igraph_vcount()` or `igraph_ecount()`. This would also let us change `igraph_mean_degree()` to just use a return value.

Reply:
I thought about this, and actually expected that you'd vote for using an error code anyway. This was one reason I chose this. The other was that `igraph_density()` also returns an error code, even though it doesn't really need it with the current implementation.

There are two ways I can imagine that a future version of these functions could fail:

 - We change the underlying graph representation and checking for loops is no longer a straightforward iteration through edges without memory allocation. But then it is likely that many more interface changes will become necessary.
 - The caching functions become more complicated and start potentially requiring memory allocation. I don't _think_ this will happen, and I would hope that it doesn't happen, but I am not 100% sure. The next major change to caching is the "deduction engine", which I _think_ wouldn't require memory allocation (I'd definitely want to avoid it given how expensive `malloc` is).

I'm on the fence about this. I'll leave the decision to you. Let me know if I should refactor.

Note that `igraph_has_loop()` can also return its result directly, but `igraph_has_multiple()` currently cannot.

Reply:
I am very slightly leaning towards keeping the error code, partly out of caution (future stability), and partly for reasons for symmetry/consistency with other functions. But I do recognize that this "convenience function" for a mean degree is not much more convenient in C than `2.0 * igraph_ecount(g) / igraph_vcount(g)` if we have to use an output parameter.